lib/amqp/Vhost.js

```diff
@@ -171,6 +172,7 @@ function Vhost(config) {
   };
 
   function createChannelPool(options) {
+    var poolOptions = _.defaultsDeep({}, { idleTimeoutMillis: 0, evictionRunIntervalMillis: 0 }, options.pool);
```

Comment:
Please take a close look at this and the related changes in the baseline config

Reply:
I think this makes sense. We don't know how long a reply will take, so don't want channels being reaped by the pool. This leads me to think we might want to either make the new behaviour `opt-in` or use a separate pool (as per confirm channels). If someone had a pool with a large max limit and had a burst of traffic, it could create a high number of channels that would never be destroyed.

I know it's a pain, but what do you think about using a different channel pool, can creating a function `borrowReplyChannel`, that was used only by publications configured with `directReplies: true` ?